gio/src/initable.rs

```diff
@@ -18,6 +23,21 @@ impl Initable {
         unsafe { object.init(cancellable)? };
         Ok(object)
     }
+
+    // rustdoc-stripper-ignore-next
+    /// Create a new instance of an initable object of the given type with the given properties.
+    ///
+    /// Similar to [`Object::with_type`] but can fail either because the object
+    /// creation failed or because `Initable::init` failed.
+    pub fn with_type(
+        type_: Type,
+        properties: &[(&str, &dyn ToValue)],
+        cancellable: Option<&impl IsA<Cancellable>>,
+    ) -> Result<Object, InitableError> {
```

Comment:
What is the point of providing the type if you have to downcast later?

Reply:
Why do you have to downcast later? And even if you have to, `type_` and the type you downcast to might not even be the same. The type you might downcast to might be just a parent class because there's no public API for the actual type.

Reply:
Ah right, never mind :) I saw that it downcast  later on the test case